infra/src/main/kotlin/fi/fta/geoviite/infra/geocoding/Geocoding.kt

```diff
@@ -736,6 +735,19 @@ fun splitRange(range: ClosedRange<TrackMeter>, splits: List<ClosedRange<TrackMet
         else maxOf(range.start, allowedRange.start)..minOf(range.endInclusive, allowedRange.endInclusive)
     }
 
+fun <T, R : Comparable<R>> getSublistForRangeInOrderedListExclusive(
+    things: List<T>,
+    range: Range<R>,
+    compare: (thing: T, rangeEnd: R) -> Int,
+): List<T> =
+    getIndexRangeForRangeInOrderedList(things, range.min, range.max, compare)?.let { indexRange ->
+        val exclusiveStart =
+            if (compare(things[indexRange.first], range.min) <= 0) indexRange.first + 1 else indexRange.first
+        val exclusiveEnd =
+            if (compare(things[indexRange.last], range.max) >= 0) indexRange.last else indexRange.last + 1
```

Comment:
Tässä varmaan kuuluisi olla `indexRange.last - 1` eikä `+ 1`, vai? Nythän tämä indeksoi rangen loppukohdan ohi, kun oletettavasti tarkoitus olisi pysyä sen sisällä

Tällaisille suht. yksinkertaisille mutta tärkeille helppereille voisi myös muuten kirjoitella ihan omiakin, puhtaasti näitä funktioita testaavia yksikkötestejä

Reply:
... mutta nyt kun tätä katson niin tuo compare-osuushan on väärin. Oikeasti tuossa halutaan tarkastaa vain case == mutta ajattelin että jos se arvo nyt jostain syystä olisi vielä yli siitä pyydetystä niin voi sen silloinkin suodattaa... ja pistin sitten > ja < väärin päin. Taidan vaihtaa nuo == merkeiksi koska tuo getIndexRangehan toimii päin honkia jos sieltä jotain muuta tulee.